### DISCUSSION TITLE
QR Code Imediato

### DISCUSSION THREAD

**Discussion by mzavarize** [bacen/pix-api#392] [General]
Boa tarde,

Alguns bancos não estão lendo o QR Code Imediato gerado por nós, o mesmo foi homologado pelo QR Tester, tem algum contato aqui dos bancos abaixo para que eu possa entrar em contato para tentar identificar o problema?
Banco Santander
Banco Itaú
Banco Inter
Peço desculpa estar usando esse espaço, mas não consegui localizar em outro local um meio de contato.

**Comment by rubenskuhl** [comment]
> Boa tarde,
> 
> Alguns bancos não estão lendo o QR Code Imediato gerado por nós, o mesmo foi homologado pelo QR Tester, tem algum contato aqui dos bancos abaixo para que eu possa entrar em contato para tentar identificar o problema?
> Banco Santander
> Banco Itaú
> Banco Inter
> Peço desculpa estar usando esse espaço, mas não consegui localizar em outro local um meio de contato.

Poderia postar um EMV aqui (de preferência colocando entre 3 ` para o GitHub não distorcer) para avaliarmos ? 
Lembrando que algumas coisas o QR-Tester ainda não testa, por exemplo x5t no JWS.

**Reply by mzavarize** [reply]
Boa tarde @rubenskuhl ,

Segue o EMV, 
'''
Estrutura EMV-TLV Básica (ID | Tamanho | Dados):
PASS "00020101021226880014br.gov.bcb.pix2566pix.credisan.net.br/pix/qr/v2/8c748929-dac3-4aa3-a2ec-2ef849077fcb520400005303986540510.005802BR5924MARCUS VINICIUS ZAVARIZE6015SAO JOAO DA BOA62070503***63045ACA" 
 0  2 "01" 
 1  2 "12" 
26 88 "0014br.gov.bcb.pix2566pix.credisan.net.br/pix/qr/v2/8c748929-dac3-4aa3-a2ec-2ef849077fcb" 
       0 14 "br.gov.bcb.pix" 
      25 66 "pix.credisan.net.br/pix/qr/v2/8c748929-dac3-4aa3-a2ec-2ef849077fcb" 
52  4 "0000" 
53  3 "986" 
54  5 "10.00" 
58  2 "BR" 
59 24 "MARCUS VINICIUS ZAVARIZE" 
60 15 "SAO JOAO DA BOA" 
62  7 "0503***" 
       5  3 "***" 
63  4 "5ACA" 
'''

No Banco do Brasil, Caixa e Nubank decodificou esse QR, por isso estou meio perdido.

Obrigado pelo apoio,

**Reply by rubenskuhl** [reply]
O EMV está certo, mas como você não colocou entre três ` (é a aspa simples para esquerda) os ```***``` sumiram, mas deu para consertar. Decodificando o JWT usando o JWT.io está dando assinatura inválida, então parece que é algum problema aí do lado de vocês mesmo.

**Reply by mzavarize** [reply]
Olá @rubenskuhl,

Fiz o teste de um payload de outro banco que foi lido pelos que estou tendo problema e ao testar no jwt.io ele também exibiu Assinatura Inválida.

**Reply by LeoBOrtigoza-zz** [reply]
Para mim aqui, o EMV não está correto, não está respeita as normas TLV (type length value). 

00020101021226880014br.gov.bcb.pix2566pix.credisan.net.br/pix/qr/v2/8c748929-dac3-4aa3-a2ec-2ef849077fcb520400005303986540510.005802BR5924MARCUS VINICIUS ZAVARIZE6015SAO JOAO DA BOA6207050363045ACA

validei no bacen, no nosso sistema e também nesse site que acho que pode ajudar você.
https://openpix.com.br/qrcode/scanner/

**Reply by mzavarize** [reply]
Obrigado pela dica @LeoBOrtigoza, vou verificar.

**Reply by rubenskuhl** [reply]
> Olá @rubenskuhl,
> 
> Fiz o teste de um payload de outro banco que foi lido pelos que estou tendo problema e ao testar no jwt.io ele também exibiu Assinatura Inválida.

Pode ser um modo de falha que eles não estejam checando. O PSP que uso gera JWTs que passam ilesos no jwt.io, e não tem problemas de aceitação.

**Reply by rubenskuhl** [reply]
> Para mim aqui, o EMV não está correto, não está respeita as normas TLV (type length value).

Pq o GitHub omitiu os ```***```

**Reply by mzavarize** [reply]
copia e cola correto '''00020101021226880014br.gov.bcb.pix2566pix.credisan.net.br/pix/qr/v2/8c748929-dac3-4aa3-a2ec-2ef849077fcb520400005303986540510.005802BR5924MARCUS VINICIUS ZAVARIZE6015SAO JOAO DA BOA62070503***63045ACA'''

**Reply by LeoBOrtigoza-zz** [reply]
Era o *** mesmo que o Git recusou, perdão! 
Realmente é um problema de certificado, o seu domínio está com os padrões de segurança correto? 

De acordo com esse manual https://www.bcb.gov.br/content/estabilidadefinanceira/cedsfn/Manual%20de%20Seguranca%20do%20PIX%20v3.2.pdf

**Reply by mzavarize** [reply]
Bom dia, analisando o log do meu servidor, quando faço a consulta dos QR Codes pelo banco Inter, Santander e Itaú, os mesmos não chegam nem a realizar a consulta do JWS no meu servidor. O Certificado CERTQRC está disponível no download pela Interface ARQ e o EMV conforme verificado está correto também, vejo que com esses dois itens corretos, a consulta deveria ser realizada ao nosso servidor para receber o JWS. Solicitei a nossa equipe de infra para verificar se estamos com algum problema de rota.

**Reply by rubenskuhl** [reply]
> Bom dia, analisando o log do meu servidor, quando faço a consulta dos QR Codes pelo banco Inter, Santander e Itaú, os mesmos não chegam nem a realizar a consulta do JWS no meu servidor. O Certificado CERTQRC está disponível no download pela Interface ARQ e o EMV conforme verificado está correto também, vejo que com esses dois itens corretos, a consulta deveria ser realizada ao nosso servidor para receber o JWS. Solicitei a nossa equipe de infra para verificar se estamos com algum problema de rota.

Notar que o JWS trafega via Internet, não via RSFN.

**Reply by mzavarize** [reply]
Olá @rubenskuhl , a questão da rota seria via internet mesmo. Obrigado.

**Reply by mzavarize** [reply]
Um detalhe que possa estar influenciando, seria que estamos com dois certificados CERTQRC, um que iríamos usar inicialmente e o que está em uso atualmente, estamos vendo para solicitar a desativação do antigo, não sei se os bancos estaria verificando só o primeiro e ignorando o segundo.

**Reply by cryptographix** [reply]
@mzavarize estou recebendo um 410 (Gone) do servidor para o payload 'pix.credisan.net.br/pix/qr/v2/8c748929-dac3-4aa3-a2ec-2ef849077fcb'

**Reply by mzavarize** [reply]
Olá @SeanWykes , esse QR Code expirou. Segue um novo para teste.
'''00020101021226880014br.gov.bcb.pix2566pix.credisan.net.br/pix/qr/v2/2c0fd452-938e-49ea-b974-e3e77720936f520400005303986540555.005802BR5924MARCUS VINICIUS ZAVARIZE6015SAO JOAO DA BOA62070503***63045887'''
Obrigado,

**Reply by mliberato** [reply]
Vale verificar se a URL utilizada corresponde à um dos domínios válidos indicados pelo arquivo de certificados subidos através da interface ARQ.
O certificado utilizado no domínio que serve o payload é que deve estar listado lá associado ao domínio. O certificado utilizado para assinar o JWS serve só para assinar o JWS (e deve estar presente no campo x5c na sua JKU).

**Reply by mzavarize** [reply]
Olá @mliberato , obrigado, um dos certificados que estão no ARQ é do host que estão sendo publicado o QRCode, este certificado está conforme o manual de segurança, sendo tipo EV com os parâmetros informados e temos o segundo certificado que assina o JWS. 
Muito obrigado,

**Reply by mzavarize** [reply]
Bom dia, estamos buscando contato nos bancos (Santander, Itaú e Inter) que estão com problema para tratarmos junto a eles a identificação do problema. 
Muito obrigado,
